### DISCUSSION TITLE
Support for webhooks?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3742] [Q&A]
I was asked last night about support for webhooks in our API.  This would allow us to signal the consumer of the API that we have updated some data.  Before I start researching this, I wanted to ask the community if anyone has successfully supported webhooks using Oqtane.

**Comment by markdav-is** [comment]
This could be accomplished by creating a module that has a config page where they can supply a URL, and then registering a job that would fire when needed to let them know when the data changes.

**Comment by sbwalker** [comment]
@markdav-is you may also want to look at IEventSubscriber:

https://www.oqtane.org/blog/!/53/oqtane-server-events

Oqtane raises events for all major actions and you can create a subscriber to consume them - and it could then POST them to a webhook.

**Comment by thabaum** [comment]
@iJungleboy It would be nice to see this topic being explained in Oqtane.Docs documentation as one of the developer guides.  A blog and/or Wiki articles along with a video to showcase this would be great as well 🚀 but at least in documentation to allow everyone a pit stop to get their engine started when heading that direction providing a clearly mapped out route on how to get there.  I am sure you have experience here as well so, I just thought you or someone up for the task may be able to put a useful guide together at some point in time.🎿
